nautobot/docs/additional-features/jobs.md

```diff
@@ -553,57 +553,94 @@ Provision of user input (`data` values) via the CLI is not supported at this tim
+  from nautobot.extras.models import JobResult, Job
+
+  def run_job_for_testing(job, data=None, commit=True, username="test-user"):
+      if data is None:
+        data = {}
+      user_model = get_user_model()
+      user, _ = user_model.objects.get_or_create(username=username, is_superuser=True, password="password")
+      job_result = JobResult.objects.create(
+          name=job.class_path,
+          obj_type=ContentType.objects.get_for_model(Job),
+          user=user,
+          job_id=uuid.uuid4(),
+      )
+      with web_request_context(user=user) as request:
+          run_job(data=data, request=request, commit=commit, job_result_pk=job_result.pk)
+      return job_result
+  ```
+- Setup the `databases` field on the test class correctly, and re-create the default Statuses on `setUp` in your test classes, because `django.test.TransactionTestCase` truncates them on every `tearDown`:
+  ```python
+  from django.apps import apps
+  from django.conf import settings
+  from django.test import TransactionTestCase
+
+  from nautobot.extras.management import populate_status_choices
+
+  class MyJobTestCase(TransactionTestCase):
+    # 'job_logs' is a proxy connection to the same (default) database that's used exclusively for Job logging
+    if "job_logs" in settings.DATABASES:
+        databases = ("default", "job_logs")
 
-        if "job_logs" in settings.DATABASES:
-            settings.DATABASES["job_logs"] = settings.DATABASES["job_logs"].copy()
-            settings.DATABASES["job_logs"]["TEST"] = {"MIRROR": "default"}
+    def setUp(self):
+        super().setUp()
+        populate_status_choices(apps, None)
+  ```
 
-    This code isn't *needed* in Nautobot 1.3.x and later, as the `"job_logs"` database will already be set up correctly for testability, but it also won't hurt anything to include.
+---
 
-A simple example of a Job test case that's implemented to be backwards-compatible with Nautobot 1.2.x might look like the following:
+A simple example of a Job test case for 1.3.x and forward might look like the following:
 
 ```python
-import uuid
-
-from django.conf import settings
-from django.contrib.contenttypes.models import ContentType
+from django.apps import apps
 from django.test import TransactionTestCase
 
-from nautobot.extras.jobs import run_job
-from nautobot.extras.models import Job, JobResult, JobLogEntry
-
-
-if "job_logs" in settings.DATABASES:
-    settings.DATABASES["job_logs"] = settings.DATABASES["job_logs"].copy()
-    settings.DATABASES["job_logs"]["TEST"] = {"MIRROR": "default"}
+from nautobot.extras.management import populate_status_choices
+from nautobot.extras.models import Job, JobLogEntry
+from nautobot.utilities.testing import run_job_for_testing
 
 
 class MyJobTestCase(TransactionTestCase):
-    if "job_logs" in settings.DATABASES:
-        databases = ("default", "job_logs")
 
+    def setUp(self):
+        super().setUp()
+        populate_status_choices(apps, None)
+    
```

Comment:
Not needed, right, now that `TransactionTestCase` itself calls `populate_status_choices`?